shared/src/question-generators/StackQueue/QueueGenerator.ts

```diff
@@ -0,0 +1,818 @@
+import { Language } from "@shared/api/Language.ts"
+import {
+  minimalMultipleChoiceFeedback,
+  MultiFreeTextFeedbackFunction,
+  MultiFreeTextFormatFunction,
+  Question,
+  QuestionGenerator,
+} from "@shared/api/QuestionGenerator.ts"
+import { serializeGeneratorCall } from "@shared/api/QuestionRouter.ts"
+import { Queue } from "@shared/question-generators/StackQueue/Queue.ts"
+import { parseArrayString } from "@shared/question-generators/StackQueue/StackGenerator.ts"
+import Random from "@shared/utils/random.ts"
+import { t, tFunction, tFunctional, Translations } from "@shared/utils/translations.ts"
+
+const translations: Translations = {
+  en: {
+    name: "Queue-Implementation using an Array",
+    description: "Correctly use queue operations",
+    solutionFreetext: `|Index|Question|Solution|\n{{0}}`,
+    performOperations: `**We perform the following operations:**{{0}}`,
+    checkFormat: "Please only enter a number.",
+    checkFormatArray: "Please only enter numbers separated by commas.",
+    getQueueInfo:
+      "**Note:** The method **getQueue()** returns the complete queue as it is as a string. If the queue is not full, the remaining elements are filled with -1.",
+    toStringInfo:
+      "**Note:** The method **toString()** only returns the part between rear and front. Every other value from the field is ignored.",
+    queueEmpty: "Currently the queue is empty.",
+    queueContainsValues: `The queue currently contains the following elements:`,
+    multipleChoiceText:
+      `Consider having a **Queue "{{0}}"**, which can store at most` +
```

Comment:
Grammatik ("having"), und hier müsste was über das Array stehen.
`Consider a **Queue "{{0}}"** implemented as a fixed array of size ...`

Reply:
Habe dies bei Stacks auch geändert zu `Consider a **Stack "{{1}}"** implemented as a dynamic/fixed array of size ...`